Sources/WebSocket.swift

```diff
@@ -248,6 +250,8 @@ open class FoundationStream : NSObject, WSStream, StreamDelegate  {
     }
     
     public func cleanup() {
+        cleanupMutex.lock()
```

Comment:
The look will need to be applied universally to protect access to the var protected here or it won't help much. I would also use an NSRecursiveLock unless you can fully understand the calling model that exists.

Reply:
I don't think recursive lock is necessary here, but I understand what you are saying about protecting all the variables that are used within the lock across the class. That said, so far in practice this has eliminated the crashes without issue. Provide a better implementation for Dalton so we can get a fix merged!